kernel/cooking.ml

```diff
@@ -251,4 +256,115 @@ let cook_constant ~hcons { from = cb; info } =
 (* let cook_constant_key = CProfile.declare_profile "cook_constant" *)
 (* let cook_constant = CProfile.profile2 cook_constant_key cook_constant *)
 
+(********************************)
+(* Discharging mutual inductive *)
+
+(* Replace
+
+     Var(y1)..Var(yq):C1..Cq |- Ij:Bj
+     Var(y1)..Var(yq):C1..Cq; I1..Ip:B1..Bp |- ci : Ti
+
+   by
+
+     |- Ij: (y1..yq:C1..Cq)Bj
+     I1..Ip:(B1 y1..yq)..(Bp y1..yq) |- ci : (y1..yq:C1..Cq)Ti[Ij:=(Ij y1..yq)]
+*)
+
+let it_mkNamedProd_wo_LetIn b d =
+  List.fold_left (fun c d -> mkNamedProd_wo_LetIn d c) b d
```

Comment:
How about moving this to Term and removing the copy in Termops?

Reply:
That would make the API asymmetrical, there are other similar functions in Termops not used in the kernel.